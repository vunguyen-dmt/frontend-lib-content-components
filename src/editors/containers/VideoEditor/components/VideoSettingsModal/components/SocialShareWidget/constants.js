export const analyticsEvents = {
  socialSharingSettingChanged: 'edx.social.video_sharing_setting.changed',
};

export default analyticsEvents;
