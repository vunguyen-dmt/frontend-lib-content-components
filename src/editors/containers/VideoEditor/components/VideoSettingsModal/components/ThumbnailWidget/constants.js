import { StrictDict } from '../../../../../../utils';

export const acceptedImgKeys = StrictDict({
  gif: '.gif',
  jpg: '.jpg',
  jpeg: '.jpeg',
  png: '.png',
  bmp: '.bmp',
  bmp2: '.bmp2',
});
export const MAX_FILE_SIZE_MB = 2000000;
export const MIN_FILE_SIZE_KB = 2000;
export const MAX_WIDTH = 1280;
export const MAX_HEIGHT = 720;
export const MIN_WIDTH = 640;
export const MIN_HEIGHT = 360;
export const ASPECT_RATIO = 1.7777777777777777777;
export const ASPECT_RATIO_ERROR_MARGIN = 0.1;
export default {
  acceptedImgKeys,
  MAX_FILE_SIZE_MB,
  MIN_FILE_SIZE_KB,
  MAX_WIDTH,
  MAX_HEIGHT,
  MIN_WIDTH,
  MIN_HEIGHT,
  ASPECT_RATIO,
  ASPECT_RATIO_ERROR_MARGIN,
};
