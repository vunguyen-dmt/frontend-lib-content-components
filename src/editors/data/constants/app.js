import { StrictDict } from '../../utils';

/* eslint-disable import/prefer-default-export */
export const blockTypes = StrictDict({
  html: 'html',
  video: 'video',
  problem: 'problem',
  // ADDED_EDITORS GO BELOW
  video_upload: 'video_upload',
  game: 'game',
});
