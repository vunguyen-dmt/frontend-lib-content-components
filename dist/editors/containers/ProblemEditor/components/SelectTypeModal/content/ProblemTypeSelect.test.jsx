import React from 'react';
import { shallow } from 'enzyme';
import { ProblemTypeKeys } from '../../../../../data/constants/problem';
import * as module from './ProblemTypeSelect';

describe('ProblemTypeSelect', () => {
  const props = {
    selected: null,
    setSelected: jest.fn(),
  };

  describe('snapshot', () => {
    test('SINGLESELECT', () => {
      expect(shallow(
        <module.ProblemTypeSelect {...props} selected={ProblemTypeKeys.SINGLESELECT} />,
      )).toMatchSnapshot();
    });
    test('MULTISELECT', () => {
      expect(shallow(
        <module.ProblemTypeSelect {...props} selected={ProblemTypeKeys.MULTISELECT} />,
      )).toMatchSnapshot();
    });
    test('DROPDOWN', () => {
      expect(shallow(
        <module.ProblemTypeSelect {...props} selected={ProblemTypeKeys.DROPDOWN} />,
      )).toMatchSnapshot();
    });
    test('NUMERIC', () => {
      expect(shallow(
        <module.ProblemTypeSelect {...props} selected={ProblemTypeKeys.NUMERIC} />,
      )).toMatchSnapshot();
    });
    test('TEXTINPUT', () => {
      expect(shallow(
        <module.ProblemTypeSelect {...props} selected={ProblemTypeKeys.TEXTINPUT} />,
      )).toMatchSnapshot();
    });
  });
});
