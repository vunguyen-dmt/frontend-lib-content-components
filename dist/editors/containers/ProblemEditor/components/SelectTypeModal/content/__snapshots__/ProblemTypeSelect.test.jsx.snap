// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProblemTypeSelect snapshot DROPDOWN 1`] = `
<Container
  style={
    Object {
      "height": "400px",
      "width": "494px",
    }
  }
>
  <SelectableBox.Set
    columns={1}
    onChange={[Function]}
    type="radio"
    value="optionresponse"
  >
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="multiplechoiceresponse"
      type="radio"
      value="multiplechoiceresponse"
    >
      Single select
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="choiceresponse"
      type="radio"
      value="choiceresponse"
    >
      Multi-select
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="optionresponse"
      type="radio"
      value="optionresponse"
    >
      Dropdown
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="numericalresponse"
      type="radio"
      value="numericalresponse"
    >
      Numerical input
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="stringresponse"
      type="radio"
      value="stringresponse"
    >
      Text input
    </SelectableBox>
  </SelectableBox.Set>
  <Button
    className="pl-0 mt-2"
    onClick={[Function]}
    variant="link"
  >
    <FormattedMessage
      defaultMessage="Advanced problem types"
      description="Button label for advance problem types option"
      id="authoring.problemEditor.problemSelect.advanceButton.label"
    />
  </Button>
</Container>
`;

exports[`ProblemTypeSelect snapshot MULTISELECT 1`] = `
<Container
  style={
    Object {
      "height": "400px",
      "width": "494px",
    }
  }
>
  <SelectableBox.Set
    columns={1}
    onChange={[Function]}
    type="radio"
    value="choiceresponse"
  >
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="multiplechoiceresponse"
      type="radio"
      value="multiplechoiceresponse"
    >
      Single select
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="choiceresponse"
      type="radio"
      value="choiceresponse"
    >
      Multi-select
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="optionresponse"
      type="radio"
      value="optionresponse"
    >
      Dropdown
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="numericalresponse"
      type="radio"
      value="numericalresponse"
    >
      Numerical input
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="stringresponse"
      type="radio"
      value="stringresponse"
    >
      Text input
    </SelectableBox>
  </SelectableBox.Set>
  <Button
    className="pl-0 mt-2"
    onClick={[Function]}
    variant="link"
  >
    <FormattedMessage
      defaultMessage="Advanced problem types"
      description="Button label for advance problem types option"
      id="authoring.problemEditor.problemSelect.advanceButton.label"
    />
  </Button>
</Container>
`;

exports[`ProblemTypeSelect snapshot NUMERIC 1`] = `
<Container
  style={
    Object {
      "height": "400px",
      "width": "494px",
    }
  }
>
  <SelectableBox.Set
    columns={1}
    onChange={[Function]}
    type="radio"
    value="numericalresponse"
  >
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="multiplechoiceresponse"
      type="radio"
      value="multiplechoiceresponse"
    >
      Single select
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="choiceresponse"
      type="radio"
      value="choiceresponse"
    >
      Multi-select
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="optionresponse"
      type="radio"
      value="optionresponse"
    >
      Dropdown
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="numericalresponse"
      type="radio"
      value="numericalresponse"
    >
      Numerical input
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="stringresponse"
      type="radio"
      value="stringresponse"
    >
      Text input
    </SelectableBox>
  </SelectableBox.Set>
  <Button
    className="pl-0 mt-2"
    onClick={[Function]}
    variant="link"
  >
    <FormattedMessage
      defaultMessage="Advanced problem types"
      description="Button label for advance problem types option"
      id="authoring.problemEditor.problemSelect.advanceButton.label"
    />
  </Button>
</Container>
`;

exports[`ProblemTypeSelect snapshot SINGLESELECT 1`] = `
<Container
  style={
    Object {
      "height": "400px",
      "width": "494px",
    }
  }
>
  <SelectableBox.Set
    columns={1}
    onChange={[Function]}
    type="radio"
    value="multiplechoiceresponse"
  >
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="multiplechoiceresponse"
      type="radio"
      value="multiplechoiceresponse"
    >
      Single select
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="choiceresponse"
      type="radio"
      value="choiceresponse"
    >
      Multi-select
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="optionresponse"
      type="radio"
      value="optionresponse"
    >
      Dropdown
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="numericalresponse"
      type="radio"
      value="numericalresponse"
    >
      Numerical input
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="stringresponse"
      type="radio"
      value="stringresponse"
    >
      Text input
    </SelectableBox>
  </SelectableBox.Set>
  <Button
    className="pl-0 mt-2"
    onClick={[Function]}
    variant="link"
  >
    <FormattedMessage
      defaultMessage="Advanced problem types"
      description="Button label for advance problem types option"
      id="authoring.problemEditor.problemSelect.advanceButton.label"
    />
  </Button>
</Container>
`;

exports[`ProblemTypeSelect snapshot TEXTINPUT 1`] = `
<Container
  style={
    Object {
      "height": "400px",
      "width": "494px",
    }
  }
>
  <SelectableBox.Set
    columns={1}
    onChange={[Function]}
    type="radio"
    value="stringresponse"
  >
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="multiplechoiceresponse"
      type="radio"
      value="multiplechoiceresponse"
    >
      Single select
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="choiceresponse"
      type="radio"
      value="choiceresponse"
    >
      Multi-select
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="optionresponse"
      type="radio"
      value="optionresponse"
    >
      Dropdown
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="numericalresponse"
      type="radio"
      value="numericalresponse"
    >
      Numerical input
    </SelectableBox>
    <SelectableBox
      aria-label="checkbox"
      className="border border-light-400 text-primary-500 shadow-none"
      id="stringresponse"
      type="radio"
      value="stringresponse"
    >
      Text input
    </SelectableBox>
  </SelectableBox.Set>
  <Button
    className="pl-0 mt-2"
    onClick={[Function]}
    variant="link"
  >
    <FormattedMessage
      defaultMessage="Advanced problem types"
      description="Button label for advance problem types option"
      id="authoring.problemEditor.problemSelect.advanceButton.label"
    />
  </Button>
</Container>
`;
